lib/portlayer/storage/vsphere/volume.go

```diff
@@ -189,8 +203,26 @@ func (v *VolumeStore) VolumeCreate(ctx context.Context, ID string, store *url.UR
 	return vol, nil
 }
 
-func (v *VolumeStore) VolumeDestroy(ctx context.Context, ID string) error {
-	return fmt.Errorf("TBD.  Not supported yet")
+func (v *VolumeStore) VolumeDestroy(ctx context.Context, vol *storage.Volume) error {
+	if err := volumesInUse(ctx, vol.ID); err != nil {
```

Comment:
Yes, we do.  In the case of the VM being off but still referring to an image, vsphere will happily let you nuke the image.  If the VM is then powered on, vsphere will return an error.

Likewise if have a VM that is on (during the in use check) and you attempt to remove that image before RM is called, and then the VM is powered off, NOOP-  user gets an error.

If a VM is off and turned on pre RM, the in use check will fail and the user will get the well crafted error.

If the VM